### PR TITLE
Miscellaneous Flake8 warnings fixed

### PR DESCRIPTION
**Pull request by josephframsay**
As part of #20508, several Python scripts have been refactored to reduce cyclomatic complexity

**To test:**
Make sure all unit and system tests pass, and code review. In addition, if this is your area, feel free to throw some data at me to test the affected scripts with, to make sure I haven't broken anything

Address part of #20508 . 

**Release Notes** 
*Does not need to be in the release notes as the change is internal*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by peterfpeterson — APPROVED** [review]
These files no longer have `python3-flake8` warnings in them.

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesQuasi.py [hunk 1]
```diff
@@ -386,6 +368,33 @@ def PyExec(self):
...
+        if not is_supported_f2py_platform():
+            unsupported_msg = "This algorithm can only be run on valid platforms." \
+                              + " please view the algorithm documentation to see" \
+                              + " what platforms are currently supported"
+            raise RuntimeError(unsupported_msg)
+
+    def fortran_background_encoding(self, setup_prog):
+        # fortran code uses background choices defined using the following numbers
+        setup_prog.report('Encoding input options')
+        if self._background == 'Sloping':
```
Comment (AndreiSavici) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesQuasi.py]: All these if statements should be `return ['Zero','Flat','Sloping'].index(self._background)`

**Review by AndreiSavici — CHANGES_REQUESTED** [review]

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesQuasi.py [hunk 2]
```diff
@@ -149,19 +144,12 @@ def PyExec(self):
...
         # convert true/false to 1/0 for fortran
-        o_el = 1 if self._elastic else 0
-        o_w1 = 1 if self._width else 0
-        o_res = 1 if self._res_norm else 0
+        o_el = int(self._elastic)
+        o_w1 = int(self._width)
+        o_res = int(self._res_norm)
 
         # fortran code uses background choices defined using the following numbers
-        setup_prog.report('Encoding input options')
```
Comment (AndreiSavici) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesQuasi.py]: Is this line supposed to be gone?

Comment (josephframsay) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesQuasi.py]: Nope - replaced now!

**Review by AndreiSavici — APPROVED** [review]

**Comment by SimonHeybrock** [comment]
The PR message still says "in progress", should this be merged or are there more changes coming?

**Comment by josephframsay** [comment]
Nope, just forgot to remove that. Everything's done on this one

**Pull request merged**